test/RToken.test.js

```diff
@@ -1854,4 +1861,148 @@ contract("RToken", accounts => {
+        // Accumulate interest
+        await doBingeBorrowing(142);
+        await expectAccount(admin, {
+            tokenBalance: "0.00000",
+            cumulativeInterest: "0.00000",
+            receivedLoan: "100.00000",
+            receivedSavings: "100.00014",
+            interestPayable: "0.00014",
+        });
+        await expectAccount(customer1, {
+            tokenBalance: "1000.00000",
+            cumulativeInterest: "0.00000",
+            receivedLoan: "0.00000",
+            receivedSavings: "0.00000",
+            interestPayable: "0.00000",
+        });
+        await expectAccount(customer2, {
+            tokenBalance: "0.00000",
+            cumulativeInterest: "0.00000",
+            receivedLoan: "0.00000",
+            receivedSavings: "0.00000",
+            interestPayable: "0.00000",
+        });
+        await expectAccount(customer3, {
+            tokenBalance: "0.00000",
+            cumulativeInterest: "0.00000",
+            receivedLoan: "900.00000",
+            receivedSavings: "900.00128",
+            interestPayable: "0.00128",
+        });
+        await expectAccount(customer4, {
+            tokenBalance: "0.00000",
+            cumulativeInterest: "0.00000",
+            receivedLoan: "0.00000",
+            receivedSavings: "0.00000",
+            interestPayable: "0.00000",
+        });
+        // owner changeHat to different recipient
+        await web3tx(rToken.createHat, "rToken.createHat for customer1 benefiting customer2 and customer4 49/51")(
+            [customer2, customer4], [49, 51], true, {
+                from: customer1
+            }
+        );
+        // Accumulate interest
+        await doBingeBorrowing(298);
+        await expectAccount(admin, {
+            tokenBalance: "0.00000",
+            cumulativeInterest: "0.00000",
+            receivedLoan: "0.00000",
+            receivedSavings: "0.00014",
+            interestPayable: "0.00014",
+        });
+        await expectAccount(customer1, {
+            tokenBalance: "1000.00000",
+            cumulativeInterest: "0.00000",
+            receivedLoan: "0.00000",
+            receivedSavings: "0.00000",
+            interestPayable: "0.00000",
+        });
+        await expectAccount(customer2, {
+            tokenBalance: "0.00000",
+            cumulativeInterest: "0.00000",
+            receivedLoan: "490.00000",
+            receivedSavings: "490.00293",
+            interestPayable: "0.00293",
+        });
+        await expectAccount(customer3, {
+            tokenBalance: "0.00000",
+            cumulativeInterest: "0.00000",
+            receivedLoan: "0.00000",
+            receivedSavings: "0.00129",
+            interestPayable: "0.00129",
+        });
+        await expectAccount(customer4, {
+            tokenBalance: "0.00000",
+            cumulativeInterest: "0.00000",
+            receivedLoan: "510.00000",
+            receivedSavings: "510.00305",
+            interestPayable: "0.00304",
```

Comment:
It seems a bit strange that the amount after the decimal point of `receivedSavings` is not the same as the one for `interestPayable`.

Reply:
```
// receivedSavingsOf

        uint256 rGross = sInternalToR(account.sInternalAmount);

// getInterestPayableOf

        uint256 rGross = sInternalToR(account.sInternalAmount);
    return rGross - account.lDebt - account.rInterest;
```

It's quite possible the lDebt is slightly larger than 510.00000